index.js

```diff
@@ -38,23 +37,15 @@ export const useRouter = () => useContext(RouterCtx);
  * Part 1, Hooks API: useRoute and useLocation
  */
 
-export const useLocation = () => {
-  const router = useRouter();
-  return router.hook(router);
-};
+const _useLocation = (router) => router.hook(router);
```

Comment:
Thanks for updating this file as well. Do you think we could give this function a different name to avoid confusion? 
Maybe `useRouterLocation` (not ideal though) or `useLocationWithinRouter`?
As a well as some short comment to explain that it's an internal version of `useLocation` that works with any router passed to it.